test/PushConstant/no_8bit_storage_pod_args.cl

```diff
@@ -0,0 +1,50 @@
+// RUN: clspv %target %s -o %t.spv -inline-entry-points -cl-std=CL3.0 -no-8bit-storage=pushconstant -no-16bit-storage=pushconstant
+// RUN: spirv-dis %t.spv -o %t.spvasm
+// RUN: FileCheck %s < %t.spvasm
+// RUN: clspv-reflection %t.spv -o %t.map
+// RUN: FileCheck --check-prefix=MAP %s < %t.map
+// RUN: spirv-val --target-env vulkan1.0 %t.spv
+
+__kernel void test_kernel(char2 c, uchar2 uc, short2 s, int2 i, __global float2 *result)
+{
+    result[0] = convert_float2(c);
+    result[1] = convert_float2(uc);
+    result[2] = convert_float2(s);
+    result[3] = convert_float2(i);
+}
+
+// CHECK-DAG: [[uchar:%[a-zA-Z0-9_]+]] = OpTypeInt 8 0
+// CHECK-DAG: [[ushort:%[a-zA-Z0-9_]+]] = OpTypeInt 16 0
+// CHECK-DAG: [[uint:%[a-zA-Z0-9_]+]] = OpTypeInt 32 0
+// CHECK-DAG: [[int_0:%[a-zA-Z0-9_]+]] = OpConstant [[uint]] 0
+// CHECK-DAG: [[int_1:%[a-zA-Z0-9_]+]] = OpConstant [[uint]] 1
+// CHECK-DAG: [[int_2:%[a-zA-Z0-9_]+]] = OpConstant [[uint]] 2
+// CHECK-DAG: [[int_3:%[a-zA-Z0-9_]+]] = OpConstant [[uint]] 3
+// CHECK-DAG: [[int_4:%[a-zA-Z0-9_]+]] = OpConstant [[uint]] 4
+// CHECK-DAG: [[v4uchar:%[a-zA-Z0-9_]+]] = OpTypeVector [[uchar]] 4
+// CHECK-DAG: [[v2uchar:%[a-zA-Z0-9_]+]] = OpTypeVector [[uchar]] 2
+// CHECK-DAG: [[v2ushort:%[a-zA-Z0-9_]+]] = OpTypeVector [[ushort]] 2
+// CHECK-DAG: [[v2int:%[a-zA-Z0-9_]+]] = OpTypeVector [[uint]] 2
+// CHECK-DAG: [[undef_var:%[a-zA-Z0-9_]+]] = OpUndef [[v4uchar]]
+// CHECK-DAG: [[struct_ptr:%[a-zA-Z0-9_]+]] = OpTypePointer PushConstant {{.*}}
+// CHECK-DAG: [[var:%[a-zA-Z0-9_]+]] = OpVariable [[struct_ptr]] PushConstant
+// CHECK-DAG: [[int_ptr:%[a-zA-Z0-9_]+]] = OpTypePointer PushConstant [[uint]]
+
+// CHECK: OpLabel
+
+// CHECK-DAG: [[gep0:%[a-zA-Z0-9_]+]] = OpAccessChain [[int_ptr]] [[var]] [[int_0]] [[int_0]]
+// CHECK-DAG: [[char_int:%[a-zA-Z0-9_]+]] = OpLoad [[uint]] [[gep0]]
+// CHECK-DAG: [[char4_var:%[a-zA-Z0-9_]+]] = OpBitcast [[v4uchar]] [[char_int]]
+// CHECK-DAG: OpVectorShuffle [[v2uchar]] [[char4_var]] [[undef_var]] 0 1
+// CHECK-DAG: OpVectorShuffle [[v2uchar]] [[char4_var]] [[undef_var]] 2 3
```

Comment:
Could you also add a check for the usage of this shuffle? I think it is important to make sure what this shuffle is used for, and it helps understand what the test is actually checking.
```
         %33 = OpVectorShuffle %v2uchar %29 %31 2 3
         %50 = OpConvertUToF %v2float %33
         %51 = OpAccessChain %_ptr_StorageBuffer_v2float %18 %uint_0 %uint_1
               OpStore %51 %50
```